.github/workflows/comment-pr.yml

```diff
@@ -0,0 +1,55 @@
+name: Comment on the pull request
+
+on:
+  workflow_run:
+    workflows: ["free-programming-books-lint"]
+    types:
+      - completed
+
+jobs:
+  upload:
+    permissions:
+      pull-requests: write
+    runs-on: ubuntu-latest
+    if: >
+      ${{ github.event.workflow_run.event == 'pull_request' &&
+      github.event.workflow_run.conclusion == 'success' }}
+    steps:
+      - name: 'Download artifact'
+        uses: actions/github-script@v6
+        with:
+          script: |
+            let allArtifacts = await github.rest.actions.listWorkflowRunArtifacts({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               run_id: context.payload.workflow_run.id,
+            });
+            let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
+              return artifact.name == "pr"
+            })[0];
+            let download = await github.rest.actions.downloadArtifact({
+               owner: context.repo.owner,
+               repo: context.repo.repo,
+               artifact_id: matchArtifact.id,
+               archive_format: 'zip',
+            });
+            let fs = require('fs');
+            fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/pr.zip`, Buffer.from(download.data));
+
+      - name: 'Unzip artifact'
+        run: unzip pr.zip
+
+      - name: 'Comment on PR'
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I think since this does not run `on: pull_request` but on another workflow completed, secrets might be available.

Reply:
Yup! Secrets should be available in `comment-pr.yml` flow because that's internal for the repo and does not trigger through the fork. (It's triggered indirectly.)

Or at least, this is to the best of my knowledge. If you're up for it, you could try to do a faulty PR to my fork on purpose, and we'll see if the action works. 🤔 No problem if you don't have time.

Reply:
Yeah I can do this tonight (US time). Will post the results here.

Reply:
I have [created a PR](https://github.com/SethFalco/free-programming-books/pull/7) on your branch. It is "awaiting workflow approvals". Let me know if I can do anything else here to test this, happy to help!

Reply:
Approved!

![](https://github.com/EbookFoundation/free-programming-books/assets/22801583/0fa6d3f7-2c16-4d36-89c0-05b7ac190314)

Reply:
Awesome, LGTM!